tests/data/ssl4eo_downstream_landsat/data.py

```diff
@@ -0,0 +1,194 @@
+years = {"tm": 2011, "etm": 2019, "oli": 2019}
+
+def create_image(path: str) -> None:
+    profile = {
+        "driver": "GTiff",
+        "dtype": "uint8",
+        "nodata": None,
+        "width": SIZE,
+        "height": SIZE,
+        "count": num_bands["_".join(path.split(os.sep)[1].split("_")[2:][:-1])],
+        "crs": CRS.from_epsg(4326),
+        "transform": Affine(
+            0.00037672803497508636,
+            0.0,
+            -109.07063613660262,
+            0.0,
+            -0.0002554026278261721,
+            47.49838726154881,
+        ),
+        "blockysize": 1,
+        "tiled": False,
+        "compress": "lzw",
+        "interleave": "pixel",
+    }
+
+    Z = np.random.randint(low=0, high=255, size=(SIZE, SIZE))
+
+    with rasterio.open(path, "w", **profile) as src:
+        for i in src.indexes:
+            src.write(Z, i)
+
+
+def create_mask(path: str) -> None:
+    profile = {
+        "driver": "GTiff",
+        "dtype": "uint8",
+        "nodata": None,
+        "width": SIZE,
+        "height": SIZE,
+        "count": 1,
+        "crs": CRS.from_epsg(4326),
+        "transform": Affine(
+            0.00037672803497508636,
+            0.0,
+            -109.07063613660262,
+            0.0,
+            -0.0002554026278261721,
+            47.49838726154881,
+        ),
+        "blockysize": 1,
+        "tiled": False,
+        "compress": "lzw",
+        "interleave": "band",
+    }
+
+    Z = np.random.randint(low=0, high=10, size=(1, SIZE, SIZE))
+
+    with rasterio.open(path, "w", **profile) as src:
+        src.write(Z)
+
+def create_img_directory(directory: str, hierarchy: FILENAME_HIERARCHY) -> None:
+    if isinstance(hierarchy, dict):
+        # Recursive case
+        for key, value in hierarchy.items():
+            if any([x in key for x in filenames.keys()]):
+                key = f"ssl4eo_l_{key}_benchmark"
+            path = os.path.join(directory, key)
+            os.makedirs(path, exist_ok=True)
+            create_img_directory(path, value)
+    else:
+        # Base case
+        for value in hierarchy:
+            path = os.path.join(directory, value)
+            create_image(path)
+
+def create_mask_directory(directory: str, hierarchy: FILENAME_HIERARCHY, mask_product: str) -> None:
+    if isinstance(hierarchy, dict):
+        # Recursive case
+        for key, value in hierarchy.items():
+            # if any([x in key for x in filenames.keys()]):
+            #     key = key.split("_")[0] # only keep toa, etm, oil
+            path = os.path.join(directory, key)
+            os.makedirs(path, exist_ok=True)
+            create_mask_directory(path, value, mask_product)
+    else:
+        # Base case
+        for value in hierarchy:
+            path = os.path.join(directory, value)
+            year = years[path.split(os.sep)[1].split("_")[2]]
+            create_mask(path.replace("all_bands", f"{mask_product}_{year}.tif"))
```

Comment:
Lot of code duplication here. If you add the masks to the FILENAME_HIERARCHY and change the few things that are different based on filename, this shouldn't be necessary.

Reply:
I tried something else before, but I thought this was more clear, even though there is code duplication.